src/patchers/SwitchPatcher.js

```diff
@@ -0,0 +1,83 @@
+import NodePatcher from './NodePatcher.js';
+import type { Editor, Node, ParseContext, Token } from './types.js';
+
+export default class SwitchPatcher extends NodePatcher {
+  constructor(node: Node, context: ParseContext, editor: Editor, expression: NodePatcher, cases: Array<NodePatcher>, alternate: ?NodePatcher) {
+    super(node, context, editor);
+    this.expression = expression;
+    this.cases = cases;
+    this.alternate = alternate;
+  }
+
+  patchAsStatement() {
+    let { expression, alternate } = this;
+    let endOfLine = this.getFirstLineBreak();
+
+    if (expression) {
+      if (!expression.isSurroundedByParentheses()) {
+        expression.insertBefore('(');
+        expression.patch();
+        expression.insertAfter(')');
+      }
+    } else {
+      this.cases.forEach((casePatcher) => casePatcher.negate());
+      this.insert(endOfLine, ' (false)');
```

Comment:
maybe just get the `switch` token instead of finding the end of the line? What if it looks like this?

```coffee
switch # test
  when a
```